### ISSUE TITLE
Options binding question

### ISSUE THREAD

**Issue by daveeel** [ManuelDeLeon/viewmodel#158]
Hi @ManuelDeLeon , I have a selection setup as:
```jade
    select($b="options:nodeList, optionsValue:id, optionsText:title, value:selectedNodeId, defaultText: '- Select -', defaultValue: ''")
    a.btn.btn-sm($b="click: removeNode, if:selectedNodeId") Remove {{selectedNodeId}}

```
```ls
    selected-node-id: ->
      (Session.get \SELECTED_ID) || ''
```
`SELECTED_ID` is set by some events in a D3 graph, which can't be accessed by VM. The desired behaviour is that the `selectedNodeId` could be set by either Session or by manually selecting the dropbox. However, when I select from the selection, `selected-node-id` won't update. Selection works if I make it like..
```ls
    selected-node-id: ''
```
Then can't set it from the D3 events. I have to use autorun:
```ls
    autorun: ->
      @selected-node-id (Session.get \SELECTED_ID)
```

So, is there a simpler way to work from both ways without autorun?

**Comment by ManuelDeLeon** [comment]
The thing is that you're trying to have a 3-way bind between the select, the view model, and the session variable. In short, ViewModel isn't designed for that scenario.

You just gave me a good reason to bring back v1's `ViewModel.byId`. That way you can just update the view model directly instead of going through a session variable intermediary.

**Comment by daveeel** [comment]
Yes, that would be very handy if we could update the VM without Session. 

For my use case, the D3 graph is an object in `on-rendered` like:
```ls
    on-rendered: !->
      nodes = nodes-cursor.fetch!
      links = links-to-d3-array db-links, nodes

      graph = new GraphManager \#graph-manager, nodes, links
      ...
```
I haven't tried, but just wondering if we could pass a VM prop via the object's parameter, and update via that reference? say:
```ls
      graph = new GraphManager \#graph-manager, nodes, links, @some-vm-ref
```

**Comment by daveeel** [comment]
Just did a quick test, and passing `@some-vm-ref` by object parameter actually works :)

**Comment by daveeel** [comment]
Just did quick test, passing vm prop via object parameter actually did what I need.

**Comment by ManuelDeLeon** [comment]
I'm glad it's working for you. That's how you used to share properties in v1 (tossing them around).

I'll still add `ViewModel.find` and `ViewModel.findOne`. I think there's a case for it.

**Comment by daveeel** [comment]
I do think passing parameters is not elegant enough. Would like to see how ViewModel.find / findOne could simplify the pattern.

Currently, I use ViewModel.byId quite often in console for debug. But need to check Id first before finding the target.

By the way, what do you think about event sending and listening in VM ? Would that be more generic for inter-VM interaction?

**Comment by ManuelDeLeon** [comment]
Pick up v2.9.0 which has find and findOne

> what do you think about message sending and listening in VM ? Would that be more generic for inter-VM interaction?

After many years the cleanest and simplest pattern I've seen is to use `ViewModel.share` to 'communicate' between view models. If you think about it you're not really sending messages from one to another, you're sharing the state of one so the other can do something with that information. `ViewModel.share` lets you do just that.

**Comment by daveeel** [comment]
Great, I shall check out .share.

By the way, I just tested .find and had the following error on console:
```
ViewModel.find('graphManager')
es5-shim.js:954 Uncaught TypeError: Cannot convert undefined or null to object(…)keys @ es5-shim.js:954_.values @ underscore.js:803ViewModel.find @ viewmodel.coffee:68(anonymous function) @ VM46707:2InjectedScript._evaluateOn @ VM46700:875InjectedScript._evaluateAndWrap @ VM46700:808InjectedScript.evaluate @ VM46700:664
```

`ViewModel.find()` is fine, and also with empty string say `ViewModel.find('')

**Comment by ManuelDeLeon** [comment]
I'll check it out and fix it.

Btw, see my phone book project on github for examples of how view models
can use share. The best example is the header with a search box and the
list of contacts filter by that input box.

**Comment by ManuelDeLeon** [comment]
Fixed on v2.9.1

**Comment by daveeel** [comment]
Works as expected :+1:
